### PR TITLE
feat #1217: Open side panel when new node is added

### PR DESCRIPTION
**Pull request by ibek**

### CODE REVIEW

File: packages/ui/src/components/Visualization/Custom/ContextMenu/ItemAddStep.tsx [hunk 1]
```diff
@@ -29,7 +46,14 @@ export const ItemAddStep: FunctionComponent<ItemAddStepProps> = (props) => {
 
     /** Update entity */
     entitiesContext.updateEntitiesFromCamelResource();
-  }, [catalogModalContext, entitiesContext, props.mode, props.vizNode]);
+
+    setTimeout(() => {
+      const result = props.vizNode.data.path
+        ? findVisualizationNode(controller.getGraph(), props.vizNode.data.path)
+        : null;
+      controller.fireEvent(SELECTION_EVENT, [result?.getNextNode()?.id]);
```
Comment (lordrip) [inline packages/ui/src/components/Visualization/Custom/ContextMenu/ItemAddStep.tsx:54]: There are a few cases in which `getNextNode` doesn't yield the desired node:
1. When adding a step in `from`
2. Preprending a step from any given node

I'll try to check if we can get the new node id directly.

**Review by apupier — CHANGES_REQUESTED** [review]
* several cases are not working:
  * When adding a step after the first element (it is called "add step" in this case), the config panel is closing
NotSelectingNewNodeWithAddStep
  * When prepending a step, the select node is not the new node but the one on which we prepended
selectingCurrentInsteadOfCreatedWhenPrepending

which I guess are linked to comment here https://github.com/KaotoIO/kaoto/pull/1563#discussion_r1796732772

File: packages/ui/src/components/Visualization/Custom/ContextMenu/ItemAddStep.tsx (hunk 1 above)
Comment (ibek) [reply packages/ui/src/components/Visualization/Custom/ContextMenu/ItemAddStep.tsx:54]: Thanks, I will wait for your changes in https://github.com/lordrip/kaoto/commit/3d668cd273d2e2de8dfa5e0526d946c213b9246e before adding more capabilities

**Comment by lordrip** [comment]
Thanks for the contribution @ibek, based on this PR, I created the helper function to identify `IVizNodes`, I'll close it for now and reopen it again after the release, when we start looking at this again.

**Pull request closed without merging**